test/gtest/ucs/test_profile.cc

```diff
@@ -164,14 +164,25 @@ void test_profile::run_profiled_code(int num_iters)
     if (num_threads() == 1) {
         profile_thread_func(&param);
     } else {
-        pthread_t threads[num_threads()];
+        std::vector<pthread_t> threads;
```

Comment:
seems like FIFO `std::queue` fits better

Reply:
why, we need a container not a fifo..

Reply:
to avoid moving elements on join and destroy, but it's minor